demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/send/view/SendView.kt

```diff
@@ -76,8 +79,10 @@ fun Send(
     sendState: SendState,
     onSend: (ZecSend) -> Unit,
     onGetProposal: (ZecSend) -> Unit,
+    onGetProposalFromUri: (String) -> Unit,
     onBack: () -> Unit,
     sendTransactionProposal: Proposal?,
+    sendTransactionProposalFromUri: Proposal?,
```

Comment:
This is weird to me. Why is there this added parameter? Once you get a proposal back, you should only need one method to send the proposal.

Reply:
Note that it is important that once a proposal has been computed and displayed to the user that it is _that exact_ proposal that is executed; you cannot recompute the proposal from the original payment request URI because the fee may change due to differences in note selection.

Reply:
ALSO, the same thing is true for proposals created from parameters, not just for proposals created from ZIP 321 URIs: once *any* proposal has been displayed to the user, that exact proposal must be executed, you can't recompute it from the parameters. If that's not currently being done, that's a bug.